power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -271,9 +316,19 @@ inline auto get_edge_weights(TransformerGraph const& graph) -> TrafoGraphEdgePro
         if (graph[e].regulated_idx == unregulated_idx) {
             continue;
         }
-        auto edge_res = std::min(vertex_distances[boost::source(e, graph)], vertex_distances[boost::target(e, graph)]);
+        // control_side, two winding or three winding, is always parallel to the
+        // source side of the transformer edge
+        auto const edge_src_rank = vertex_distances[boost::source(e, graph)];
+        auto const edge_tgt_rank = vertex_distances[boost::target(e, graph)];
+        auto const edge_res = std::min(edge_src_rank, edge_tgt_rank);
```

Comment:
In two winding trafo, the edge is always pointing to the control side; in three winding trafo edges, the edges are always pointing parallel to the control side: meaning that for delta configuration ABC, the following situations can happen:
```
|  Tap  | Control |         All edges       |
---------------------------------------------
|   A   |    A    | [B->A], [C->A], [B<->C] |
|   A   |    B    | [A->B], [A->C], [B<->C] |
|   A   |    C    | [A->B], [A->C], [B<->C] |
|   B   |    A    | [B->A], [C<->A], [B->C] |
|   B   |    B    | [A->B], [C<->A], [C->B] |
|   B   |    C    | [B->A], [C<->A], [B->C] |
|   C   |    A    | [A<->B], [C->A], [C->B] |
|   C   |    B    | [A<->B], [C->A], [C->B] |
|   C   |    C    | [A<->B], [A->C], [A->B] |
```

Reply:
This piece of comment is added directly to the code as comment for better readibility.